### ISSUE TITLE
Toggle old alignment behavior for equations

### ISSUE THREAD

**Issue by fredrik-johansson** [sphinx-doc/sphinx#2370]
Previously, if you put two equations inside one math block

```
.. math ::

    A

    B
```

both would show up centered. Due to recent changes in sphinx (which puts them in an align environment), they now render right-aligned. I can get back to two centered equations using

```
.. math ::

    A

.. math ::

    B
```

but this is less convenient. Is there any way to restore the old behavior?

**Comment by fredrik-johansson** [comment]
BTW, the workaround to get two centered equations is not perfect. Although it gives similar output to the old behavior with HTML+MathJax, it creates an extra vertical space in latexpdf output.

**Comment by tk0miya** [comment]
Indeed. It comes from #2254. I reproduce this with following code:

```
Single equation in single math directive:

.. math ::

    A

Multiple equations in single math directive:

.. math ::

    A

    B

Multiple equations in multiple math directives:

.. math ::

    A

.. math ::

    B
```

Indeed, the equations are slightly misaligned.
![2016-03-24 21 55 02](https://cloud.githubusercontent.com/assets/748828/14017704/a2fc4e68-f20c-11e5-9d48-3eec5e45e068.png)

**Comment by jfbu** [comment]
Perhaps there is some more up-to-date info but I have already found this : http://tex.stackexchange.com/a/8142/4686, which says

> for some reason, a one-line "aligned" is not properly centered. this appears to be a bug, and i am entering it as such in the amsmath bugs list.

Anyway, things are produced like

``` latex
\begin{align*}\begin{aligned}
\begin{split}A\end{split}\\
\begin{split}B\end{split}\end{aligned}\end{align*}
```

and I am not sure about `split` inside `aligned` inside `align*`. The `split` environment is designed for use in full-width environments `align`, `gather`,  or `equation`. The amsmath documentation doesn't have a single example of `split` inside `aligned`. Where would the `&` and `\\` which `split` uses come from ?

**Comment by jfbu** [comment]
I obtain centered results with

``` latex
Single equation in single math directive:
\begin{align*}\!\begin{aligned}
\begin{split}\boxed{A}\end{split}\\\end{aligned}\end{align*}
Multiple equations in single math directive:
\begin{align*}\!\begin{aligned}
\begin{split}\boxed{A}\end{split}\\
\begin{split}\boxed{B}\end{split}\\\end{aligned}\end{align*}

\noindent\rule{.5\linewidth}{3pt}%
\kern-.2pt 
\smash{\rule[-5cm]{.4pt}{10cm}}%
\kern-.2pt 
\rule{.5\linewidth}{6pt}

Multiple equations in multiple math directives:
\begin{align*}\!\begin{aligned}
\begin{split}\boxed{A}\end{split}\\\end{aligned}\end{align*}\begin{align*}\begin{aligned}
\begin{split}\boxed{B}\end{split}\\\end{aligned}\end{align*}
```

![capture d ecran 2016-03-25 a 17 16 08](https://cloud.githubusercontent.com/assets/2589111/14048244/5281b382-f2ad-11e5-86f2-d3ff0da4718a.png)

The recipe appears to be to use `\!\begin{aligned}.....\\\end{aligned}`, hence an extra `\!` before `\begin{aligned}` and an extra `\\` before `\end{aligned}`. Not tested apart from the above.

(and I still don't understand the reason for `split`)

**Comment by jfbu** [comment]
In my previous comment notice that the bottom boxed B is not centered but I forgot the `\!` before `\begin{aligned}` for the last one. If one adds it, the bottom boxed B is centered.

**Comment by tk0miya** [comment]
Thank you for comment. It looks very nice!

The `aligned` also comes from #2254 . It is used to wrap whole of equations and give only one equation-number label.
But I don't know why `aligned` and `split` is used at same time. I did not care that.
@xuhdev Do you remember the reason?

**Comment by xuhdev** [comment]
I think it's OK to remove `split` inside `aligned`. I don't remember the exact reason, but I feel like that I added `split` inside `aligned` for semantic purpose only: `split` splits single equations. If it causes any issues, we can remove it.

According to @jfbu 's comments, we should use different environments for single equations (`split` inside `equation`) and multiple equations (`align`).

**Comment by tk0miya** [comment]
I fixed this at 39396a4 and 3206ccbae0c73958a2e120b5d5d02a00150b776d.

Finally I use `equation` and `split` for single equation and `align` and `aligned` for multiple equations.
If you find another problem, please let me know.

Thanks!
